wasm/module.go

```diff
@@ -333,6 +335,20 @@ type NameAssoc struct {
 	Name  string
 }
 
+// NewNameMap constructs a new NameMap from the given name to index map and returns it ordered by index ascending.
+func NewNameMap(m map[string]Index) NameMap {
```

Comment:
I will delete this and build it iteratively as I looked at the call sites and we can